### ISSUE TITLE
ViroCore Release Build Errors

### ISSUE THREAD

**Issue by darshanio** [NativeVision/virocore#167]
- [ ] Review the documentation: https://virocore.viromedia.com/
    **_Document site not working_**
  - [ ] Search for existing issues: https://github.com/viromedia/virocore/issues
    **_No existing issue_**
  - [ ] Use the latest ViroCore release: https://virocore.viromedia.com/docs/releases
    **_Release site not working_**
  - [ ] Turn off Instant Run in Android Studio
    **_Document site not working_**

## Environment
Please provide the following information about your environment:
1. OS: Mac
2. Version: ViroCore 1.21.0 (main branch)
3. Device(s): trying to build the library (Google Pixel 6a)

## Description
Is the Viromedia site down? Why can't we see any documents?
Is this project being maintained?
jitpack.io does not show the package for the version 1.21.0

The build fails when tried using the following gradle command
```bash
./gradlew :virocore:assembleRelease
```

The configurations are as following:
**Java**
openjdk 21.0.3 2024-04-16
OpenJDK Runtime Environment Homebrew (build 21.0.3)
OpenJDK 64-Bit Server VM Homebrew (build 21.0.3, mixed mode, sharing)

**Android SDK**
Android: API 34
Installed SDK build tools:
1) 35.0.0-rc4
2) 34.0.0

Build logs:
```log
* What went wrong:
Execution failed for task ':virocore:compileReleaseJavaWithJavac'.
> Could not resolve all files for configuration ':virocore:androidJdkImage'.
   > Failed to transform core-for-system-modules.jar to match attributes {artifactType=_internal_android_jdk_image, org.gradle.libraryelements=jar, org.gradle.usage=java-runtime}.
      > Execution failed for JdkImageTransform: /Users/darshanio/Library/Developer/Xamarin/android-sdk-macosx/platforms/android-33/core-for-system-modules.jar.
         > Error while executing process /opt/homebrew/Cellar/openjdk/21.0.3/libexec/openjdk.jdk/Contents/Home/bin/jlink with arguments {--module-path /Users/darshanio/.gradle/caches/transforms-3/d846070216edce3934e051ac24fc5ac1/transformed/output/temp/jmod --add-modules java.base --output /Users/darshanio/.gradle/caches/transforms-3/d846070216edce3934e051ac24fc5ac1/transformed/output/jdkImage --disable-plugin system-modules}

```

**Comment by darshanio** [comment]
When tried with OpenJDK 17, the build fails for virocore with following build failure logs.

<details>
<summary>Build logs</summary>

```
> Task :virocore:compileReleaseJavaWithJavac FAILED
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1751: error: class, interface, enum, or record expected
    private native Node nativeCreateNode();
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1752: error: class, interface, enum, or record expected
    native int  nativeGetUniqueIdentifier(long nodeReference);
           ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1753: error: class, interface, enum, or record expected
    private native void nativeDestroyNode(long nodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1754: error: class, interface, enum, or record expected
    private native void nativeAddChildNode(long nodeReference, long childNodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1755: error: class, interface, enum, or record expected
    private native void nativeRemoveFromParent(long nodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1756: error: class, interface, enum, or record expected
    protected native void nativeRemoveAllChildNodes(long nodeReference);
                     ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1757: error: class, interface, enum, or record expected
    private native void nativeSetHierarchicalRendering(long nodeReference, boolean hierarchicalRendering);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1758: error: class, interface, enum, or record expected
    private native void nativeSetName(long nodeReference, String name);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1759: error: class, interface, enum, or record expected
    private native void nativeSetGeometry(long nodeReference, long geoReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1760: error: class, interface, enum, or record expected
    private native void nativeClearGeometry(long nodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1761: error: class, interface, enum, or record expected
    private native void nativeSetPosition(long nodeReference, float x, float y, float z);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1762: error: class, interface, enum, or record expected
    private native void nativeSetRotationEuler(long nodeReference, float x, float y, float z);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1763: error: class, interface, enum, or record expected
    private native void nativeSetRotationQuaternion(long nodeReference, float x, float y, float z, float w);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1764: error: class, interface, enum, or record expected
    private native void nativeSetScale(long nodeReference, float x, float y, float z);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1765: error: class, interface, enum, or record expected
    private native void nativeSetRotationPivot(long nodeReference, float x, float y, float z);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1766: error: class, interface, enum, or record expected
    private native void nativeSetScalePivot(long nodeReference, float x, float y, float z);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1767: error: class, interface, enum, or record expected
    private native void nativeUpdateWorldTransforms(long nodeReference, long parentReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1768: error: class, interface, enum, or record expected
    private native void nativeStartUpdateAtomicUmbrellaBounds(long nodeRef);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1769: error: class, interface, enum, or record expected
    private native float[] nativeUpdateAtomicUmbrellaBounds(long nodeToUpdateRef, long nodeRef, float[] transform);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1770: error: class, interface, enum, or record expected
    private native void nativeEndUpdateAtomicUmbrellaBounds(long nodeRef);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1771: error: class, interface, enum, or record expected
    private native void nativeSetOpacity(long nodeReference, float opacity);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1772: error: class, interface, enum, or record expected
    private native void nativeSetVisible(long nodeReference, boolean visible);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1773: error: class, interface, enum, or record expected
    private native void nativeSetRenderingOrder(long nodeReference, int renderingOrder);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1774: error: class, interface, enum, or record expected
    private native void nativeSetDragType(long nodeReference, String dragType);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1775: error: class, interface, enum, or record expected
    private native void nativeSetDragPlanePoint(long nodeReference, float[] planePoint);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1776: error: class, interface, enum, or record expected
    private native void nativeSetDragPlaneNormal(long nodeReference, float[] planeNormal);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1777: error: class, interface, enum, or record expected
    private native void nativeSetDragMaxDistance(long nodeReference, float maxDistance);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1778: error: class, interface, enum, or record expected
    private native void nativeSetIgnoreEventHandling(long nodeReference, boolean visible);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1779: error: class, interface, enum, or record expected
    private native void nativeSetHighAccuracyEvents(long nodeReference, boolean enabled);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1780: error: class, interface, enum, or record expected
    private native void nativeSetTransformBehaviors(long nodeReference, String[] transformBehaviors);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1781: error: class, interface, enum, or record expected
    private native void nativeSetEventDelegate(long nodeReference, long eventDelegateRef);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1782: error: class, interface, enum, or record expected
    private native long nativeSetTransformDelegate(long nodeReference, double throttlingWindow);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1783: error: class, interface, enum, or record expected
    private native void nativeRemoveTransformDelegate(long nodeReference, long mNativeTransformDelegate);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1784: error: class, interface, enum, or record expected
    private native float[] nativeGetPosition(long nodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1785: error: class, interface, enum, or record expected
    private native float[] nativeGetScale(long nodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1786: error: class, interface, enum, or record expected
    private native float[] nativeGetRotationEuler(long nodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1787: error: class, interface, enum, or record expected
    private native float[] nativeGetRotationQuaternion(long nodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1788: error: class, interface, enum, or record expected
    private native float[] nativeGetBoundingBox(long boundingBox);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1789: error: class, interface, enum, or record expected
    private native float[] nativeGetBoundingBoxLocal(long boundingBox);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1790: error: class, interface, enum, or record expected
    private native String[] nativeGetAnimationKeys(long nodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1791: error: class, interface, enum, or record expected
    private native void nativeSetTag(long nodeReference, String tag);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1792: error: class, interface, enum, or record expected
    private native void nativeSetCamera(long nodeReference, long cameraReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1793: error: class, interface, enum, or record expected
    private native void nativeClearCamera(long nodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1794: error: class, interface, enum, or record expected
    private native void nativeAddLight(long nodeReference, long lightReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1795: error: class, interface, enum, or record expected
    private native void nativeRemoveLight(long nodeReference, long lightReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1796: error: class, interface, enum, or record expected
    private native void nativeRemoveAllLights(long nodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1797: error: class, interface, enum, or record expected
    private native void nativeAddSound(long nodeReference, long soundReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1798: error: class, interface, enum, or record expected
    private native void nativeRemoveSound(long nodeReference, long soundReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1799: error: class, interface, enum, or record expected
    private native void nativeRemoveAllSounds(long nodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1800: error: class, interface, enum, or record expected
    private native void nativeSetParticleEmitter(long nodeRef, long particleRef);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1801: error: class, interface, enum, or record expected
    private native void nativeSetFixedParticleEmitter(long nodeRef, long particleRef);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1802: error: class, interface, enum, or record expected
    private native void nativeRemoveParticleEmitter(long nodeRef);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1803: error: class, interface, enum, or record expected
    private native float[] nativeConvertLocalPositionToWorldSpace(long nodeReference, float x, float y, float z);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1804: error: class, interface, enum, or record expected
    private native float[] nativeConvertWorldPositionToLocalSpace(long nodeReference, float x, float y, float z);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1805: error: class, interface, enum, or record expected
    private native float[] nativeGetWorldTransform(long nodeReference);
                   ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1806: error: class, interface, enum, or record expected
    protected native void nativeSetLightReceivingBitMask(long nodeReference, int bitMask, boolean recursive);
                     ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1807: error: class, interface, enum, or record expected
    protected native void nativeSetShadowCastingBitMask(long nodeReference, int bitMask, boolean recursive);
                     ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1815: error: class, interface, enum, or record expected
    private WeakReference<TransformListener> mTransformListener = null;
            ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1816: error: class, interface, enum, or record expected
    private long mNativeTransformDelegate = INVALID_REF;
            ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1838: error: class, interface, enum, or record expected
    public final void onPositionUpdate(float x, float y, float z) {
                 ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1841: error: class, interface, enum, or record expected
        }
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1852: error: class, interface, enum, or record expected
    public void setTransformListener(TransformListener transformListener, double distanceFilter) {
           ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1855: error: class, interface, enum, or record expected
        }
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1857: error: class, interface, enum, or record expected
    }
    ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1862: error: class, interface, enum, or record expected
    public void removeTransformListener() {
           ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1865: error: class, interface, enum, or record expected
            nativeRemoveTransformDelegate(mNativeRef, mNativeTransformDelegate);
            ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1866: error: class, interface, enum, or record expected
            mNativeTransformDelegate = INVALID_REF;
            ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1867: error: class, interface, enum, or record expected
        }
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1875: error: class, interface, enum, or record expected
    public static NodeBuilder<? extends Node, ? extends NodeBuilder> builder() {
                  ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1877: error: class, interface, enum, or record expected
    }
    ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1890: error: class, interface, enum, or record expected
        if (mParent != null) {
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1892: error: class, interface, enum, or record expected
            if (parent != null) {
            ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1894: error: class, interface, enum, or record expected
            }
            ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1897: error: class, interface, enum, or record expected
        for (Node child : mChildren) {
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1899: error: class, interface, enum, or record expected
        }
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1908: error: class, interface, enum, or record expected
        if (root == null) {
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1910: error: class, interface, enum, or record expected
        } else {
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1912: error: class, interface, enum, or record expected
        }
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1920: error: class, interface, enum, or record expected
        updateUmbrellaBounds(this, true, new float[] { 1, 0, 0, 0,
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1924: error: class, interface, enum, or record expected
        nativeEndUpdateAtomicUmbrellaBounds(mNativeRef);
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1926: error: class, interface, enum, or record expected
        for (Node child : mChildren) {
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1928: error: class, interface, enum, or record expected
        }
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1935: error: class, interface, enum, or record expected
    private void updateUmbrellaBounds(final Node nodeBeingUpdated, boolean skipParent, float[] transform) {
                                            ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1938: error: class, interface, enum, or record expected
        }
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:1941: error: class, interface, enum, or record expected
        }
        ^
/Users/darshanio/projects/virocore/android/sharedCode/src/main/java/com/viro/core/Node.java:2278: error: class, interface, enum, or record expected
}
^
86 errors

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':virocore:compileReleaseJavaWithJavac'.
> Compilation failed; see the compiler error output for details.

```

</details>

**Comment by darshanio** [comment]
@robertjcolley any suggestions?

**Comment by doranteseduardo** [comment]
I found the problem, I will make a release including the fix and also the ViroVideo fix from #154
